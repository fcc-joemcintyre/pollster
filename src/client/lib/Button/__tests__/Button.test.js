/* eslint-env jest */
import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import { Button } from '..';

const theme = {
  buttonPrimaryColor: '#ffffff',
  buttonPrimaryBG: '#0000ff',
  buttonColor: '#ffffff',
  buttonBG: '#00007f',
};

test ('Button default render', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <Button />
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('Button renders primary', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <Button primary />
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('Button renders custom colors', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <Button fg='#123456' bg='#234567' />
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('Button renders transparent bg', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <Button fg='#123456' bg='transparent' />
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});
