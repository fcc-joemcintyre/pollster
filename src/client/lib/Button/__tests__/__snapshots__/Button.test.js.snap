// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button default render 1`] = `
.c0 {
  font-size: 16px;
  padding: 6px 12px;
  border-radius: 4px;
  cursor: pointer;
  color: #ffffff;
  background-color: #00007f;
  border: 1px solid #00007f;
}

.c0:hover {
  background-color: #000066;
}

<button
  className="c0"
/>
`;

exports[`Button renders custom colors 1`] = `
.c0 {
  font-size: 16px;
  padding: 6px 12px;
  border-radius: 4px;
  cursor: pointer;
  color: #123456;
  background-color: #234567;
  border: 1px solid #234567;
}

.c0:hover {
  background-color: #0a2c4e;
}

<button
  className="c0"
/>
`;

exports[`Button renders primary 1`] = `
.c0 {
  font-size: 16px;
  padding: 6px 12px;
  border-radius: 4px;
  cursor: pointer;
  color: #ffffff;
  background-color: #0000ff;
  border: 1px solid #0000ff;
}

.c0:hover {
  background-color: #0000e6;
}

<button
  className="c0"
/>
`;

exports[`Button renders transparent bg 1`] = `
.c0 {
  font-size: 16px;
  padding: 6px 12px;
  border-radius: 4px;
  cursor: pointer;
  color: #123456;
  background-color: transparent;
  border: 1px solid transparent;
}

.c0:hover {
  background-color: #e6e6e6;
}

<button
  className="c0"
/>
`;
